components/match2/wikis/rocketleague/match_summary.lua

```diff
@@ -6,144 +6,187 @@
 -- Please see https://github.com/Liquipedia/Lua-Modules to contribute
 --
 
+local Class = require('Module:Class')
 local Logic = require('Module:Logic')
 local Lua = require('Module:Lua')
+local Table = require('Module:Table')
 local VodLink = require('Module:VodLink')
+local Json = require('Module:Json')
+local Abbreviation = require('Module:Abbreviation')
+local String = require('Module:StringUtils')
 
 local DisplayHelper = Lua.import('Module:MatchGroup/Display/Helper', {requireDevIfEnabled = true})
 local MatchGroupUtil = Lua.import('Module:MatchGroup/Util', {requireDevIfEnabled = true})
+local MatchSummary = Lua.import('Module:MatchSummary/Base', {requireDevIfEnabled = true})
 local OpponentDisplay = Lua.import('Module:OpponentDisplay', {requireDevIfEnabled = true})
 
+local _GREEN_CHECK = '[[File:GreenCheck.png|14x14px|link=]]'
+local _NO_CHECK = '[[File:NoCheck.png|link=]]'
+local _TIMEOUT = '[[File:Cooldown_Clock.png|14x14px|link=]]'
+local _OCTANE_PREFIX = '[[File:Octane_gg.png|14x14px|link=http://octane.gg/matches/'
+local _OCTANE_SUFFIX = '|Octane matchpage]]'
+
 local _TBD_ICON = mw.ext.TeamTemplate.teamicon('tbd')
 
-local CustomMatchSummary = {}
+-- Custom Caster Class
+local Casters = Class.new(
+	function(self)
+		self.root = mw.html.create('div')
+			:addClass('brkts-popup-comment')
+			:css('white-space','normal')
+			:css('font-size','85%')
+		self.casters = {}
+	end
+)
 
-function CustomMatchSummary.getByMatchId(args)
-	local match = MatchGroupUtil.fetchMatchForBracketDisplay(args.bracketId, args.matchId)
+function Casters:addCaster(caster)
+	if not Logic.isEmpty(caster) then
+		table.insert(self.casters, '[[' .. caster .. ']]')
+	end
+	return self
+end
 
-	local wrapper = mw.html.create('div')
-		:addClass('brkts-popup')
-
-	local function renderOpponent(opponentIndex)
-		return OpponentDisplay.BlockOpponent({
-			flip = opponentIndex == 1,
-			opponent = match.opponents[opponentIndex],
-			overflow = 'wrap',
-			teamStyle = 'short',
-		})
-			:addClass(match.opponents[opponentIndex].type ~= 'solo'
-				and 'brkts-popup-header-opponent'
-				or 'brkts-popup-header-opponent-solo-with-team')
-	end
-
-	local function renderSoloOpponentTeam(opponentIndex)
-		local opponent = match.opponents[opponentIndex]
-		if opponent.type == 'solo' then
-			local teamExists = mw.ext.TeamTemplate.teamexists(opponent.template or '')
-			local display = teamExists
-				and mw.ext.TeamTemplate.teamicon(opponent.template, match.date)
-				or _TBD_ICON
-			return mw.html.create('div'):wikitext(display)
-				:addClass('brkts-popup-header-opponent-solo-team')
-		else
-			return ''
-		end
+function Casters:create()
+	return self.root
+		:node('<b>Caster' .. (#self.casters > 1 and 's' or '') .. ':</b><br>')
+		:node(table.concat(self.casters, ', '))
+end
+
+-- Custom Header Class
+local Header = Class.new(
```

Comment:
Why can't /Base's header be used?

Reply:
because RL has several changes to it:
- adding team icons for solo opponents in some cases (hence using different css classes already and having to add those icons)
- them wanting a "bestof" display instead of score (also added the option for it to be both displayed if they want to start using Template:SingleMatch)

Reply:
> > adding team icons for solo opponents in some cases (hence using different css classes already and having to add those icons)
> 
> Seems like supporting this makes sense in base, or am I missing something? (not that used to Solo oppos)

idk if any other wiki has it
sc2 (and afaik sc) do not

> > them wanting a "bestof" display instead of score (also added the option for it to be both displayed if they want to start using Template:SingleMatch)
> 
> Can /Base be made more flexible so it can support both the current style and the RL requested style?

would need a second set of score functions in the class

also the base version has `:css('justify-content', 'center')` set to the root of the header
i think that would fuck over the display on RL
so would have to be able to "deactivate" that too

Reply:
We can figure it out later